website/docs/lifecycle-apis.md

```diff
@@ -7,11 +7,129 @@ _This section is a work in progress._
+
+## `async contentLoaded({content, actions})`
+
+Plugins should use the data loaded in `loadContent` and construct the pages/routes that consume the loaded data.
+
+### `content`
+
+`contentLoaded` will be called _after_ `loadContent` is done, the return value of `loadContent()` will be passed to `contentLoaded` as `content`.
+
+### `actions`
+
+`actions` contain two functions:
+
+- `addRoute(config: RouteConfig): void`
+- `createData(name: string, data: Object): Promise<string>`
+
+where `RouteConfig` is an object with the necessary data to configure a route to add to the website:
+
+```js
+interface RouteConfig {
+  path: string;
+  component: string;
+  modules?: RouteModule;
+  routes?: RouteConfig[];
+  exact?: boolean;
+}
+
+interface RouteModule {
+  [module: string]: Module | RouteModule | RouteModule[];
+}
+```
+
+Example `addRoute` call:
+
+```js
+addRoute({
+  path: permalink,
+  component: blogPostComponent,
+  exact: true,
+  modules: {
+    content: source,
+    metadata: metadataPath,
+    prevItem: prevItem && prevItem.metadataPath,
+    nextItem: nextItem && nextItem.metadataPath,
+  },
+});
+```
+
+And `createData` takes a file name relative to to your plugin's directory, a string for the `JSON.stringify` result of your data, and will return a path to the module which you may then use as the path to items in your `RouteModule`. The modules will be loaded when the related pages are loaded following our optimizations according to the [PRPL pattern](https://developers.google.com/web/fundamentals/performance/prpl-pattern/).
+
+## `configureWebpack(config, isServer, utils)`
+
+Modifies the internal webpack config. If the return value is a JavaScript object, it will be merged into the final config using [`webpack-merge`](https://github.com/survivejs/webpack-merge). If it is a function, it will be called and receive `config` as the first argument and an `isServer` flag as the argument argument.
+
+### `config`
+
+`configureWebpack` is called with `config` generated according to client/server build. You may treat this as the base config to be merged with.
+
+### `isServer`
+
+`configureWebpack` will be called both in server build and in client build. The server build receives `true` and the client build receives `false` as `isServer`.
+
+### `utils`
+
+The initial call to `configureWebpack` also receives a util object consists of three functions:
+
+- `getStyleLoaders(isServer: boolean, cssOptions: {[key: string]: any}): Loader[]`
+- `getCacheLoader(isServer: boolean, cacheOptions?: {}): Loader | null`
+- `getCacheLoader(isServer: boolean, cacheOptions?: {}): Loader | null`
+
+You may use them to return your webpack configures conditionally.
+
+Example:
+
+```js
+configureWebpack(config, isServer, {getBabelLoader, getCacheLoader}) {
+  const {rehypePlugins, remarkPlugins, truncateMarker} = options;
+  return {
+    module: {
+      rules: [
+        {
+          test: /(\.mdx?)$/,
+          include: [contentPath],
```

Comment:
can we simplify this part. it looks so advanced

Reply:
I think we can just remove this line. WDYT?

Reply:
oops i mean we can change the example to be more simple. eg: writing .less loader

Reply:
Feel free to make the change